include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -709,13 +731,344 @@ namespace Differentiation
+       * represent the local degree-of-freedom values.
+       *
+       * @param[in] dof_values A vector field associated with local
+       * degree-of-freedom values on the current finite element. These define
+       * the values of all independent variables. When considering taped AD
+       * numbers with branching functions, to avoid potential issues with branch
+       * switching it may be a good idea to choose these values close to those
+       * that will be later evaluated and linearized around.
+       *
+       * @note The input value type must correspond to this class's number type.
+       * So if this class is templated on type double then the number type
+       * associated with value must also be of type double.
+       *
+       * @note For taped AD numbers, this operation is only valid in recording mode.
+       */
+      void
+      register_dof_values(const std::vector<scalar_type> &dof_values);
+
+      /**
+       * Register the complete set of independent variables $\mathbf{X}$ that
+       * represent the local degree-of-freedom values.
+       *
+       * @param[in] values A global field from which the values of all
+       * independent variables will be extracted. This typically will be the
+       * solution vector around at which point a residual vector is to be
+       * computed and around which linearization is to occur.
+       * When considering taped AD numbers with branching functions, to avoid
+       * potential issues with branch switching it may be a good idea to choose
+       * these values close to those that will be later evaluated and linearized
+       * around.
+       * @param[in] local_dof_indices A vector of degree-of-freedom indices from
+       * which to extract the local degree-of-freedom values.
+       *
+       * @note For taped AD numbers, this operation is only valid in recording mode.
+       */
+      template <typename VectorType>
+      void
+      register_dof_values(
+        const VectorType &                                  values,
+        const std::vector<dealii::types::global_dof_index> &local_dof_indices);
+
+      /**
+       * Returns the complete set of degree-of-freedom values as represented by
+       * auto-differentiable numbers. These are the independent
+       * variables $\mathbf{X}$ about which the solution is linearized.
+       *
+       * It is indicated to the AD library that operations performed with these
+       * numbers are to be tracked, so they are considered "sensitive"
+       * variables. The values of the components of the returned object are
+       * initialized to the values set with register_independent_variable().
+       *
+       * @return An object of auto-differentiable type numbers representing the
+       * local degree-of-freedom values.
+       *
+       * @note For taped AD numbers, this operation is only valid in recording mode.
+       */
+      const std::vector<ad_type> &
+      get_sensitive_dof_values();
```

Comment:
maybe say in the documentation that this is the set of variables with which one would then perform computations, and based on which one can then extract both the value of the function and its derivatives with the member functions below?